File: gateway/src/main/java/bank/internalgateway/gateway/messaging/DeliveryHeaders.java
package bank.internalgateway.gateway.messaging;

/**
 * HTTP header names used when delivering events to downstream services.
 */
public final class DeliveryHeaders {

    public static final String DELIVERY_ENVELOPE = "X-Delivery-Envelope";
    public static final String EVENT_ID = "X-Event-Id";
    public static final String EVENT_TYPE = "X-Event-Type";
    public static final String IDENTITY_ENVELOPE = "X-Identity-Envelope";
    public static final String CORRELATION_ID = "X-Correlation-Id";

    private DeliveryHeaders() {
    }
}

File: dotnet-gateway/src/InternalGateway.DotNet/Messaging/DeliveryHeaders.cs
namespace InternalGateway.DotNet.Messaging;

public static class DeliveryHeaders
{
    public const string DeliveryEnvelope = "X-Delivery-Envelope";
    public const string EventId = "X-Event-Id";
    public const string EventType = "X-Event-Type";
    public const string IdentityEnvelope = "X-Identity-Envelope";
    public const string CorrelationId = "X-Correlation-Id";
    public const string DemoSubjectId = "X-Demo-Subject-Id";
    public const string DemoOrganizationId = "X-Demo-Organization-Id";
}
